guides/common/modules/proc_rex-pull-based-performance-impact-considerations.adoc

```diff
@@ -0,0 +1,24 @@
+[id="REX_pull_based_performance_impact_considerations_{context}"]
+= REX pull-based transport performance impact considerations
+
+When {ProjectServer} is configured to use REX pull-based transport for remote execution jobs using "Script" provider, {SmartProxyServer} sends a notification about new job to the client machines via MQTT. This notification does not include actual workload (actual script) the client is supposed to execute. Once a client receives a message about a new remote execution job, it queries the {SmartProxyServer} for actual payload. As the client executes the job, it periodically sends outputs of the job to the capsule, further increasing the number of requests to the {SmartProxyServer}.
+
+These requests to the {SmartProxyServer} together with high concurrency allowed by MQTT protocol can cause exhaustion of {SmartProxyServer} available connections and some requests might fail, making some child tasks of remote execution jobs to hang. This also depends on actual job payload, as some jobs are causing additional load to {ProjectServer}, making it compete for resources if you are using integrated {SmartProxyServer}.
+
+To avoid this, configure your {ProjectServer} and {SmartProxyServer} deployment appropriately with the following parameters:
+
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+mqtt_ttl - Time interval in seconds given to the host to pick up the job before considering the job undelivered. 
+mqtt_rate_limit - Number of jobs that are allowed to run at the same time
+mqtt_resend_interval - Time interval in seconds at which the notification should be re-sent to the host until the job is picked up or canceled
+----
+
+In order to tune these parameters you can follow the following procedure on the {ProjectServer}:
+
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+{foreman-installer} --foreman-proxy-plugin-remote-execution-script-mqtt-rate-limit (value) --foreman-proxy-plugin-remote-execution-script-mqtt-resend-interval (value) --foreman-proxy-plugin-remote-execution-script-mqtt-ttl (value)
+----
+
+{SmartProxyServer} logs are in /var/log/foreman-proxy/proxy.log. {SmartProxyServer} uses Webrick HTTP server (no httpd or Puma involved), so there is no simple way to increase its capacity
```

Comment:
```suggestion
{SmartProxyServer} logs are in `/var/log/foreman-proxy/proxy.log`.
{SmartProxyServer} uses Webrick HTTP server (no httpd or Puma involved), so there is no simple way to increase its capacity.
```

If you want to mention paths, please use backticks.

I'd also drop things that are not used such as Puma.